File: solutions/csharp/resistor-color/1/ResistorColor.cs
using System;

public static class ResistorColor
{
    enum ColorEnum {
        black = 0, brown, red, orange, yellow, 
        green, blue, violet, grey, white
    }

    public static int ColorCode(string color)
    {
        var val = Enum.Parse(typeof(ColorEnum), color);
        return (int)val;
    }

    public static string[] Colors()
    {
        return Enum.GetNames(typeof(ColorEnum));
    }
}

File: solutions/c/resistor-color/2/resistor_color.c
#include "resistor_color.h"

int color_code(resistor_band_t c) 
{
    return (int)c;
}

resistor_band_t *colors(){
    static resistor_band_t arr[10];

    for(resistor_band_t i = BLACK;i <= WHITE; i++)
        arr[i] = i;
    
    return arr;
}